collection/collection/src/commonTest/kotlin/androidx/collection/SparseArrayCompatTest.kt

```diff
@@ -284,6 +285,7 @@ internal class SparseArrayCompatTest {
     }
 
     @Test
+    @Ignore // Broken on Kotlin/JS.
```

Comment:
Is there any more detail that can be provided about why this is broken?

Is there a bug in the JS implementation? Or a bug in the test?

Reply:
It’s a bug in the implementation. It doesn’t fail when it should.

```
AssertionError: Expected an exception of class IndexOutOfBoundsException to be thrown, but was completed successfully.
	at protoOf.fail_zdvzi9(/Volumes/Development/androidx/GitHub/compose-multiplatform-core/out/collection/collection/build/compileSync/js/test/testDevelopmentExecutable/kotlin/jsMainSources/kotlin/kotlin/test/JsImpl.kt:23)
	at protoOf.fail_o3vfxl(/Volumes/Development/androidx/GitHub/compose-multiplatform-core/out/collection/collection/build/compileSync/js/test/testDevelopmentExecutable/kotlin/jsMainSources/kotlin/kotlin/test/DefaultJsAsserter.kt:71)
	at <global>.checkResultIsFailure(/Volumes/Development/androidx/GitHub/compose-multiplatform-core/out/collection/collection/build/compileSync/js/test/testDevelopmentExecutable/kotlin/jsMainSources/kotlin/kotlin/test/JsImpl.kt:30)
	at protoOf.valueAt_outOfBounds_muiswf(/Volumes/Development/androidx/GitHub/compose-multiplatform-core/out/collection/collection/build/compileSync/js/test/testDevelopmentExecutable/kotlin/commonMainSources/kotlin/kotlin/test/Assertions.kt:619)
	at <global>.test_fun$SparseArrayCompatTest_test_fun$valueAt_outOfBounds_test_fun_bvl1cv(compose-multiplatform-core-collection-collection-test.2895540250.js:46757)
```

I am not prepared to take on this side-quest at the moment, however!